File: MultiWaveDecoder/jaad/mp4/boxes/impl/meta/GenreBox.java
package net.sourceforge.jaad.mp4.boxes.impl.meta;

import java.io.IOException;
import net.sourceforge.jaad.mp4.MP4InputStream;
import net.sourceforge.jaad.mp4.boxes.BoxTypes;
import net.sourceforge.jaad.mp4.boxes.FullBox;
import net.sourceforge.jaad.mp4.boxes.Utils;

public class GenreBox extends FullBox {

	private string languageCode, genre;

	public GenreBox() {
		super("Genre Box");
	}

	@Override
	public void decode(MP4InputStream in) throws IOException {
		//3gpp or iTunes
		if(parent.getType()==BoxTypes.USER_DATA_BOX) {
			super.decode(in);
			languageCode = Utils.getLanguageCode(in.readBytes(2));
			final byte[] b = in.readTerminated((int) getLeft(in), 0);
			genre = new string(b, MP4InputStream.UTF8);
		}
		else readChildren(in);
	}

	public string getLanguageCode() {
		return languageCode;
	}

	public string getGenre() {
		return genre;
	}
}

File: MultiWaveDecoder/aac/mp4/boxes/impl/meta/GenreBox.cs
using System.Text;

// ReSharper disable InconsistentNaming

namespace MultiWaveDecoder
{
  public sealed class GenreBox : FullBox
  {
    string languageCode, genre;

    public GenreBox() : base("Genre Box") { }

    public override void decode(MP4InputStream inStream)
    {
      // 3gpp or iTunes
      if (parent.getType() == BoxType.USER_DATA_BOX)
      {
        base.decode(inStream);
        languageCode = BoxUtils.getLanguageCode(inStream.readBytes(2));
        var b = inStream.readTerminated((int)getLeft(inStream), 0);
        genre = Encoding.UTF8.GetString(b);
      }
      else readChildren(inStream);
    }

    public string getLanguageCode()
    {
      return languageCode;
    }

    public string getGenre()
    {
      return genre;
    }
  }
}
